ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -74,6 +75,7 @@ protected QueryMetadata(final String statementString,
 
   public void registerQueryStateListener(final QueryStateListener queryStateListener) {
     this.queryStateListener = Optional.of(queryStateListener);
+    queryStateListener.onChange(kafkaStreams.state(), kafkaStreams.state());
```

Comment:
To confirm my understanding: the addition of this line is the reason why the QueryStateListener can be initialized with a state of `"-"` (and consequently no longer needs `kafkaStreams` as a constructor input), since this line causes the initial state of the listener to be set correctly.

Reply:
Yeah, that's right. 

I could have passed `null` for the old value, (it's not even used), which may have being more intuitive, but I don't like passing `null`s around in general.